doc/guides/device_mgmt/smp_groups/smp_group_1.rst

```diff
@@ -0,0 +1,401 @@
+    | ``OP`` | ``Group ID`` | ``Command ID`` |
+    +========+==============+================+
+    | ``2``  | ``1``        |  ``1``         |
+    +--------+--------------+----------------+
+
+CBOR data of request command:
+
+.. code-block:: none
+
+    {
+        {
+            (str,opt)"image"    : (uint)
+            (str,opt)"len"      : (uint)
+            (str)"off"          : (uint)
+            (str,opt)"sha"      : (str)
+            (str,opt)"data"     : (byte str)
+            (str,opt)"upgrade"  : (bool)
+        }
+    }
+
+where
+
+.. table::
+    :align: center
+
+    +-----------------------+---------------------------------------------------+
+    | "image"               | optional image number                             |
+    +-----------------------+---------------------------------------------------+
+    | "len"                 | optional length of image, it only appears in the  |
+    |                       | first packet of request, where "off" is 0         |
+    +-----------------------+---------------------------------------------------+
+    | "off"                 | offset of image chunk the request carries         |
+    +-----------------------+---------------------------------------------------+
+    | "sha"                 | string identifying update session; although name  |
+    |                       | suggest it might be SHA it is actually any string |
+    +-----------------------+---------------------------------------------------+
+    | "data"                | optional image data                               |
+    +-----------------------+---------------------------------------------------+
+    | "upgrade"             | optional flag that states that only upgrade       |
+    |                       | should be allowed, so if version of uploaded      |
+    |                       | software is lower then already on device, the     |
+    |                       | image update should be rejected                   |
+    |                       | (unused by Zephyr at this time)                   |
+    +-----------------------+---------------------------------------------------+
+
+.. note::
+    There is no field representing size of chunk that is carried as "data" because
+    that information is embedded within "data" field itself.
+
+The mcumgr library uses "sha" field to tag ongoing update session, to be able
+to continue it in case when it gets broken.
+If library gets request with "off" equal Zero it checks stored "sha" within its
+state and if it matches it will respond to update client application with
+offset that it should continue with.
+
+The "len" and "sha" fields should only appear in first request of a session.
+
+The "image" image number is not needed at all when update is for single application
+device, so there is only one image to target, or when first image is target
+in multi-image configuration.  The field appears in first request of a session
```

Comment:
Double space. Also, the info about "image" be present only with off=0 is duplicated (assuming my previous comment).

Reply:
Done